packages/cli/src/constructs/heartbeat-check.ts

```diff
@@ -0,0 +1,68 @@
+import { Check, CheckProps } from './check'
+import { Session } from './project'
+import { DateTime } from 'luxon'
+
+type unitType = 'seconds' | 'minutes' | 'hours' | 'days'
+
+export interface Heartbeat {
+  expression?: string
+  period: number
+  periodUnit: unitType
+  grace: number
+  graceUnit: unitType
+}
+
+export interface HeartbeatCheckProps extends CheckProps {
+  expression?: string
+  period: number
+  periodUnit: unitType
+  grace: number
+  graceUnit: unitType
+}
+
+function _customPeriodGraceValidation (period: number, periodUnit: unitType, grace: number, graceUnit: unitType) {
+  const now = DateTime.now()
+  const addedTimePeriod = now.plus({ [periodUnit]: period })
+  const addedGracePeriod = now.plus({ [graceUnit]: grace })
+
+  const MAX_PERIOD_GRACE_DAYS = 365
+  const MIN_PERIOD_SECONDS = 30
+
+  if (
+    addedTimePeriod.diff(now, 'days').days > MAX_PERIOD_GRACE_DAYS ||
+    addedTimePeriod.diff(now, 'seconds').seconds < MIN_PERIOD_SECONDS
+  ) {
+    throw new Error('Period must be between 30 seconds and 365 days.')
+  }
+
+  if (addedGracePeriod.diff(now, 'days').days > MAX_PERIOD_GRACE_DAYS) {
+    throw new Error('Grace must be less than 365 days.')
+  }
+}
+
+export class HeartbeatCheck extends Check {
+  heartbeat: Heartbeat
+
+  constructor (logicalId: string, props: HeartbeatCheckProps) {
+    super(logicalId, props)
+
+    _customPeriodGraceValidation(props.period, props.periodUnit, props.grace, props.graceUnit)
+
+    this.heartbeat = {
```

Comment:
Do we need this extra `heartbeat` object?

Reply:
Yes we have validation where we need the normal check properties + a heartbeat object when a check is of type `HEARTBEAT`.

Reply:
Yes we have validation where we need the normal check properties + a heartbeat object when a check is of type HEARTBEAT.